var/spack/repos/builtin/packages/libqrencode/package.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack import *
+
+
+class Libqrencode(AutotoolsPackage):
+    """libqrencode - a fast and compact QR Code encoding library."""
+
+    homepage = "https://fukuchi.org/works/qrencode/"
+    url      = "https://github.com/fukuchi/libqrencode/archive/v4.1.1.tar.gz"
+    git      = "https://github.com/fukuchi/libqrencode.git"
+
+    maintainers = ['cessenat']
+
+    version('master', branch='master')
+    version('4.1.1', sha256='5385bc1b8c2f20f3b91d258bf8ccc8cf62023935df2d2676b5b67049f31a049c')
+
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    # We assume a reasonably recent libtool is necessary
+    depends_on('libtool@2.4.2:', type='build')
+    depends_on('m4', type='build')
+    depends_on('pkgconfig', type='build')
+    # https://fukuchi.org/works/qrencode/ requires libpng-dev
+    depends_on('libpng@1.2.0:', type=('build'))
```

Comment:
In Spack model `build` dependencies are roughly build tools and `link` are dependencies needed at compilation / link time (so most of the time a library, including header only libraries). Therefore I would do:
```suggestion
    depends_on('libpng@1.2.0:', type='link')
```